/**
 * Test that, if the socket factory throws an IOE, it properly propagates
 * to the client.
 */
@Test(timeout = 60000)
public void testSocketFactoryException() throws IOException {
    SocketFactory mockFactory = mock(SocketFactory.class);
    doThrow(new IOException("Injected fault")).when(mockFactory).createSocket();
    Client client = new Client(LongWritable.class, conf, mockFactory);
    InetSocketAddress address = new InetSocketAddress("127.0.0.1", 10);
    try {
        call(client, RANDOM.nextLong(), address, conf);
        fail("Expected an exception to have been thrown");
    } catch (IOException e) {
        assertTrue(e.getMessage().contains("Injected fault"));
    } finally {
        client.stop();
    }
}
